File: src/main/java/org/example/eventhub/dto/event/EventCreateRequest.java
package org.example.eventhub.dto.event;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import org.example.eventhub.dto.location.LocationCreateRequest;
import org.example.eventhub.enums.EventStatus;

public record EventCreateRequest(
        @NotBlank(message = "Title обязателен") String title,
        String description,
        @NotNull(message = "dateTime обязателен") LocalDateTime dateTime,
        LocationCreateRequest location,
        @NotNull(message = "capacity обязательна") @Positive(message = "capacity должна быть положительна")
                Integer capacity,
        BigDecimal price,
        EventStatus eventStatus) {}

File: src/main/kotlin/org/example/eventhub/dto/event/EventCreateRequest.kt
package org.example.eventhub.dto.event

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import org.example.eventhub.dto.location.LocationCreateRequest
import org.example.eventhub.enums.EventStatus
import java.math.BigDecimal
import java.time.LocalDateTime

data class EventCreateRequest(
    val title: @NotBlank(message = "Title обязателен") String,
    val description: String?,
    val dateTime: @NotNull(message = "dateTime обязателен") LocalDateTime,
    val location: LocationCreateRequest?,
    val capacity: @NotNull(message = "capacity обязательна") @Positive(message = "capacity должна быть положительна") Int,
    val price: BigDecimal?,
    val eventStatus: EventStatus?
)
